src/superphot_plus/ztf_transient_fit.py

```diff
@@ -143,26 +165,25 @@ def create_prior(cube):
         np.ndarray
             Updated array of parameters.
         """
+        # Compute the truncated Gaussian distribution for all values at once.
+        tg_vals = truncnorm.ppf(cube, tg_a, tg_b, loc=prior_mean, scale=prior_std)
 
-        cube[0] = max_flux * 10 ** (
-            trunc_gauss(cube[0], *PRIOR_A)
-        )  # log-uniform for A from 1.0x to 16x of max flux
-        cube[1] = trunc_gauss(cube[1], *PRIOR_BETA)  # beta UPDATED, looks more Lorentzian so widened by 1.5x
-        cube[2] = 10 ** trunc_gauss(cube[2], *PRIOR_GAMMA)  # very broad Gaussian temporary solution for gamma
-        max_flux_loc = tdata[np.argmax(fdata[bdata == "r"] - np.abs(ferrdata[bdata == "r"]))]
-        cube[3] = trunc_gauss(cube[3], np.amin(tdata) - 50.0, np.amax(tdata) + 50.0, max_flux_loc, 20.0)  # t0
-        cube[4] = 10 ** (trunc_gauss(cube[4], *PRIOR_TAU_RISE))  # taurise, UPDATED
-        cube[5] = 10 ** (trunc_gauss(cube[5], *PRIOR_TAU_FALL))  # tau fall UPDATED
-        cube[6] = 10 ** (trunc_gauss(cube[6], *PRIOR_EXTRA_SIGMA))  # lognormal for extrasigma, UPDATED
+        cube[0] = max_flux * 10 ** (tg_vals[0])  # log-uniform for A from 1.0x to 16x of max flux
+        cube[1] = tg_vals[1]  # beta UPDATED, looks more Lorentzian so widened by 1.5x
+        cube[2] = 10 ** (tg_vals[2])  # very broad Gaussian temporary solution for gamma
+        cube[3] = tg_vals[3]
+        cube[4] = 10 ** (tg_vals[4])  # taurise, UPDATED
+        cube[5] = 10 ** (tg_vals[5])  # tau fall UPDATED
+        cube[6] = 10 ** (tg_vals[6])  # lognormal for extrasigma, UPDATED
 
         # green band
-        cube[7] = trunc_gauss(cube[7], *PRIOR_A_g)  # A UPDATED
-        cube[8] = trunc_gauss(cube[8], *PRIOR_BETA_g)  # beta UPDATED
-        cube[9] = trunc_gauss(cube[9], *PRIOR_GAMMA_g)  # gamma, GAUSSIAN not Lorentzian
-        cube[10] = trunc_gauss(cube[10], *PRIOR_T0_g)  # t0 UPDATED
-        cube[11] = trunc_gauss(cube[11], *PRIOR_TAU_RISE_g)  # taurise UPDATED, Gaussian
-        cube[12] = trunc_gauss(cube[12], *PRIOR_TAU_FALL_g)  # taufall UPDATED
-        cube[13] = trunc_gauss(cube[13], *PRIOR_EXTRA_SIGMA_g)  # extra sigma UPDATED, Gaussian
+        cube[7] = tg_vals[7]  # A UPDATED
+        cube[8] = tg_vals[8]  # beta UPDATED
+        cube[9] = tg_vals[9]  # gamma, GAUSSIAN not Lorentzian
+        cube[10] = tg_vals[10]  # t0 UPDATED
+        cube[11] = tg_vals[11]  # taurise UPDATED, Gaussian
+        cube[12] = tg_vals[12]  # taufall UPDATED
+        cube[13] = tg_vals[13]  # extra sigma UPDATED, Gaussian
```

Comment:
Are we guaranteed that the input cube has len==14? If so, could we just return tg_vals instead of copying the values over? e.g.

```
tg_vals = truncnorm...
tg_vals[0 = max_flux * 10 ** tg_vals[0]  # log-uniform for A from 1.0x to 16x of max flux
# beta unchanged
tg_vals[2] = 10 ** tg_vals[2]  # very broad Gaussian temporary solution for gamma
## t0 unchagned
tg_vals[4] = 10 ** tg_vals[4]  # taurise, UPDATED
tg_vals[5] = 10 ** tg_vals[5]  # tau fall UPDATED
tg_vals[6] = 10 ** tg_vals[6]

## g band unchanged
```

Reply:
I think we are and I was originally going to implement it the way you suggested. However, I am not sure if dynesty's NestedSampler is using the fact that cube is modified in-place during the function.